yt_dlp/extractor/safari.py

```diff
@@ -117,39 +106,42 @@ class SafariIE(SafariBaseIE):
     _UICONF_ID = '29375172'
 
     def _real_extract(self, url):
+        query = {}
         mobj = self._match_valid_url(url)
 
         reference_id = mobj.group('reference_id')
         if reference_id:
-            video_id = reference_id
-            partner_id = self._PARTNER_ID
-            ui_id = self._UICONF_ID
+            query['flashvars[referenceId]'] = video_id = reference_id
+            query['wid'] = '_%s' % self._PARTNER_ID
+            query['uiconf_id'] = self._UICONF_ID
         else:
             video_id = '%s-%s' % (mobj.group('course_id'), mobj.group('part'))
 
             webpage, urlh = self._download_webpage_handle(url, video_id)
 
             mobj = re.match(self._VALID_URL, urlh.geturl())
             reference_id = mobj.group('reference_id')
-            if not reference_id:
-                reference_id = self._search_regex(
-                    r'data-reference-id=(["\'])(?P<id>(?:(?!\1).)+)\1',
-                    webpage, 'kaltura reference id', group='id')
-            partner_id = self._search_regex(
+            if reference_id:
+                query['flashvars[referenceId]'] = reference_id
+            else:
+                try:
+                    query['flashvars[referenceId]'] = self._search_regex(
+                        r'data-reference-id=(["\'])(?P<id>(?:(?!\1).)+)\1',
+                        webpage, 'kaltura reference id', group='id')
+                except RegexNotFoundError as e:
+                    self.write_debug(e.orig_msg)
+                    query['entry_id'] = self._search_regex(
+                        r'data-entry-id=(["\'])(?P<id>(?:(?!\1).)+)\1',
+                        webpage, 'kaltura entry id', group='id')
```

Comment:
```suggestion
                query['flashvars[referenceId]'] = self._search_regex(
                    r'data-reference-id=(["\'])(?P<id>(?:(?!\1).)+)\1', webpage,
                    'kaltura reference id', group='id', default='')
                self.write_debug(f'Unable to extract reference id: {e.orig_msg}')
                query['entry_id'] = self._search_regex(
                    r'data-entry-id=(["\'])(?P<id>(?:(?!\1).)+)\1', webpage, 'kaltura entry id', group='id')
```
(untested)

Reply:
https://github.com/yt-dlp/yt-dlp/pull/3536#discussion_r859900924
Only reason I stuck with this because when not logged in falls back using `entry_id` in kaltura api query, so `Unable to extract kaltura reference id` message indicates that for maintainer.

Otherwise omitting that line can make it much simpler.